src/telemetry.ts

```diff
@@ -21,173 +21,684 @@ import {
+         */
+        kernelSessionId: string;
+        /**
+         * Whether this resource is using the active Python interpreter or not.
+         */
+        isUsingActiveInterpreter?: boolean;
+        /**
+         * Found plenty of issues when starting kernels with conda, hence useful to capture this info.
+         */
+        pythonEnvironmentType?: EnvironmentType;
+        /**
+         * A key, so that rest of the information is tied to this. (hash)
+         */
+        pythonEnvironmentPath?: string;
+        /**
+         * Found plenty of issues when starting Conda Python 3.7, Python 3.7 Python 3.9 (in early days when ipykernel was not up to date)
+         */
+        pythonEnvironmentVersion?: string;
+        /**
+         * Total number of python environments.
+         */
+        pythonEnvironmentCount?: number;
+        /**
+         * Comma delimited list of hashed packages & their versions.
+         */
+        pythonEnvironmentPackages?: string;
+        /**
+         * Whether kernel was started using kernel spec, interpreter, etc.
+         */
+        kernelConnectionType?: KernelConnectionMetadata['kind'];
+        /**
+         * Language of the kernel connection.
+         */
+        kernelLanguage: string;
+        /**
+         * This number gets reset after we attempt a restart or change kernel.
+         */
+        interruptCount?: number;
+        /**
+         * This number gets reset after change the kernel.
+         */
+        restartCount?: number;
+        /**
+         * Number of times starting the kernel failed.
+         */
+        startFailureCount?: number;
+        /**
+         * Number of times the kernel was changed.
+         */
+        switchKernelCount?: number;
+        /**
+         * Total number of kernel specs in the kernel spec list.
+         */
+        kernelSpecCount: number;
+        /**
+         * Total number of interpreters in the kernel spec list.
+         */
+        kernelInterpreterCount: number;
+        /**
+         * Total number of live kernels in the kernel spec list.
+         */
+        kernelLiveCount: number;
+        /**
+         * Whether this was started by Jupyter extension or a 3rd party.
+         */
+        actionSource: KernelActionSource;
+        /**
+         * Whether we managed to capture the environment variables or not.
+         * In the case of conda environments, `false` would be an error condition, as we must have env variables for conda to work.
+         */
+        capturedEnvVars?: boolean;
+    }>;
+
+export class IEventNamePropertyMapping {
+    'ERROR': TelemetryEventInfo<
+        Partial<
+            {
+                failureCategory: 'methodException';
+                /**
+                 * Name of the method in the extension that threw the exception.
+                 */
+                failureSubCategory: string;
+            } & TelemetryErrorProperties
+        >
+    > = {
+        owner: 'donjayamanne',
```

Comment:
Types and GDPR info for properties now defined in one place and its strongly typed,
hence there's no way we can miss anything.

Reply:
Will go through this section in standup.